packages/validator/src/services/prepareBeaconProposer.ts

```diff
@@ -29,26 +37,46 @@ export class PrepareBeaconProposerService {
     clock.runEverySlot(this.prepareBeaconProposer);
   }
 
-  private prepareBeaconProposer = async (epoch: Epoch): Promise<void> => {
+  private prepareBeaconProposer = async (slot: Slot): Promise<void> => {
     await Promise.all([
       // Run prepareBeaconProposer immediately for all known local indices
-      this.api.validator
-        .prepareBeaconProposer(this.getProposerData(this.indicesService.getAllLocalIndices()))
+      this.getProposerData(slot, this.indicesService.getAllLocalIndices())
+        .then((proposerData) => this.api.validator.prepareBeaconProposer(proposerData))
         .catch((e: Error) => {
-          this.logger.error("Error on prepareBeaconProposer", {epoch}, e);
+          this.logger.error("Error on prepareBeaconProposer", {slot}, e);
         }),
 
       // At the same time fetch any remaining unknown validator indices, then poll duties for those newIndices only
       this.indicesService
         .pollValidatorIndices()
-        .then((newIndices) => this.api.validator.prepareBeaconProposer(this.getProposerData(newIndices)))
+        .then((newIndices) => this.getProposerData(slot, newIndices))
+        .then((proposerData) => this.api.validator.prepareBeaconProposer(proposerData))
         .catch((e: Error) => {
-          this.logger.error("Error on poll indices and prepareBeaconProposer", {epoch}, e);
+          this.logger.error("Error on poll indices and prepareBeaconProposer", {slot}, e);
         }),
     ]);
   };
 
-  private getProposerData(indices: number[]): ProposerPreparationData[] {
-    return indices.map((validatorIndex) => ({validatorIndex, feeRecipient: this.defaultSuggestedFeeRecipient}));
-  }
+  private getProposerData = async (slot: Slot, indices: number[]): Promise<ProposerPreparationData[]> => {
+    const proposerData: ProposerPreparationData[] = [];
+    for (const validatorIndex of indices) {
+      const pubkeyHex = this.indicesService.index2pubkey.get(validatorIndex);
+      if (!pubkeyHex) throw Error(`Pubkey lookup failure for validatorIndex=${validatorIndex}`);
```

Comment:
Give some whitespace and breathing room to the function body

```suggestion
      if (!pubkeyHex) {
        throw Error(`Pubkey lookup failure for validatorIndex=${validatorIndex}`);
      }
```